deeplake/core/dataset/dataset.py

```diff
@@ -1760,6 +1761,7 @@ def diff(
     def _populate_meta(self, verbose=True):
         """Populates the meta information for the dataset."""
         if dataset_exists(self.storage):
+            dataset_validate(self.storage)
```

Comment:
does this add impactful overhead for loading datasets? (our dataset loading is very slow in general, making it worse could be a further problem)

Reply:
It's a good question. I didn't think it would, because I was seeing self.storage as a LRUCache so my understanding was that it wouldn't need to re-fetch the files.

But, at one point I had implemented `dataset_exists(storage, validate=False)` as a single function. I ended up breaking it up since I imagined the both functions getting more complex over time and didn't want that complexity to start intertwining in a single function. But, from a performance standpoint perhaps it's worth keeping them together?

Reply:
Re-looking/thinking, I don't think we even need the new `dataset_validate` function. The reason I had added it was feeling like we used the `datastore_exists` function both to know if it's OK to create a dataset in a directory AND whether it is OK to try to load the dataset in a given directory. So I had made the new `validate` function to handle the "is it OK to load the dataset" path.

But: the load function is really the best place to be handling any "can't load this, it's invalid" logic. And really just knowing that it's a dataset-looking directory and not obviously just a directory of images or something is all we need there too.

So I took the function back out and tried doing a load on a broken dataset directory and still get a useful error message of `FileNotFoundError: [Errno 2] No such file or directory: 'test1/dataset_meta.json'` from the load which seems good enough and further simplifies this PR.